app/main.py

```diff
@@ -1,2 +1,86 @@
+from typing import Any
+
+import copy
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.length = 0
+        self.hash_table: list = [None] * 8
+        self.len_hash_tabel = len(self.hash_table)
+
+    def __len__(self) -> int:
+        return self.length
+
+    def get_cell_index(self, key: Any) -> int:
+        return hash(key) % self.len_hash_tabel
+
+    def resize_hash_tabel(self) -> None:
+        old_tabel = copy.deepcopy(self.hash_table)
+        self.hash_table = self.len_hash_tabel * [None] * 2
+        self.len_hash_tabel = len(self.hash_table)
+        self.length = 0
+        for cell in old_tabel:
+            if cell is not None:
+                created_key, _, created_value = cell
+                self.__setitem__(created_key, created_value)
+        del old_tabel
+
+    def threshold(self) -> int:
+        return int((2 / 3) * self.len_hash_tabel)
+
+    def check_threshold(self) -> bool:
+        return self.length < self.threshold()
+
+    def __setitem__(self, key: Any, value: Any) -> None:
+        if not self.check_threshold():
```

Comment:
The condition `if not self.check_threshold()` should be `if self.check_threshold()` to correctly trigger resizing when the threshold is exceeded.